src/components/Notifications/EventLog/usePrimaryToolbarFilterConfigWrapper.tsx

```diff
@@ -0,0 +1,180 @@
+            const bundle = indexOfBundle !== -1 && indexOfBundle !== undefined ? (bundles as Schemas.Facet[])[indexOfBundle] : undefined;
+
+            const bundleDisplayName = bundle?.displayName;
+            const applicationChips = (!bundle?.children ?
+                [
+                    {
+                        name: 'Loading',
+                        value: filterBundle,
+                        isRead: true
+                    }
+                ]
+                :
+                (bundle.children.length !== 0 ?
+                    bundle?.children?.map(application => ({
+                        name: application.displayName,
+                        value: application.name,
+                        isRead: true
+                    }))
+                    :
+                    [
+                        {
+                            name: bundle.displayName,
+                            value: bundle.name,
+                            isRead: true
+                        }
+                    ]
+                )
+            );
+
+            return {
+                bundleId: filterBundle,
+                category: bundleDisplayName || `${filterBundle} Loading...`,
+                chips: (filters.application as string[])?.map(filterApplication => {
+                    const appDisplayName = bundle?.children?.find(application => application.name === filterApplication)?.displayName;
+                    return {
+                        name: appDisplayName as string,
+                        value: filterApplication,
+                        isRead: true
+                    };
+                }) || applicationChips
+            };
+        }) || [];
+
+        return eventLogCustomFilters;
+    }, []);
+
+    React.useEffect(() => {
+        setCustomFilters(mapToEventLogCustomFilter(filters, bundles.length !== 0 ? bundles : undefined));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [ bundles, mapToEventLogCustomFilter ]);
+
+    toolbarConfig.filterConfig.items[1] = useMemo(() => {
+        return {
+            label: 'Application',
+            type: 'custom',
+            filterValues: {
+                children: <EventLogTreeFilter
+                    groups={ bundles }
+                    placeholder={ 'Filter by Application' }
+                    filters={ customFilters }
+                    updateFilters={ setCustomFilters }
+                />
+            }
+        } as any;
+    }, [ bundles, customFilters ]);
+
+    toolbarConfig.activeFiltersConfig.filters = React.useMemo(() => {
+        const activeFilters = toolbarConfig.activeFiltersConfig.filters;
+        const nonCustomFilters = activeFilters.filter(activeFilter => !!activeFilter && !(activeFilter as EventLogCustomFilter).bundleId);
+        return nonCustomFilters.concat(customFilters);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [ filters, customFilters ]);
```

Comment:
missing `toolbarConfig.activeFiltersConfig.filters` .

Reply:
Do not remember exactly why I didn't add this one, I did light testing and it seems that everything still works fine with it appended. In theory, 'activeFiltersConfig.filters' would only ever change due to 'filters' or 'customFilters' so this dependency should not have an effect on it, so it would not hurt to add it.